### PR TITLE
Fixed typo in email validation regular expression

### PR DESCRIPTION
**Pull request by maxnordlund**
There was two extraneous backslashes in the user_regex that essentially disabled the second range. This also contains a few more examples courtesy of https://en.wikipedia.org/wiki/Email_address

For completeness all examples from Wikipedia are included, but the ones that still fail have been commented out.

This is a small fix, but see #8081 for the future overhaul.

### CODE REVIEW

**Comment by timgraham** [comment]
Hi, I don't think qualifies for a backport to 1.10 based on our [supported versions policy](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions). For future reference, it's better to only include test cases related to the fix in a particular commit. Additional test coverage not related to a fix can be added in a separate commit. Thanks.

**Comment by maxnordlund** [comment]
It's easy to remove those out commented lines, but I thought it might be good to include them for reference. All others are relevant to this fix.

How come it doesn't qualify? I do think this fixes a fairly major bug, since the regex didn't do it's job before, right?

**Comment by charettes** [comment]
@maxnordlund please take the time to read the [docs](https://docs.djangoproject.com/en/dev/internals/release-process/#supported-versions) Tim linked to.

> Patches applied to the master branch must also be applied to the last feature release branch, to be released in the next patch release of that feature series, when they fix critical problems:
> 
> 1. Security issues.
> 2. Data loss bugs.
> 3. Crashing bugs.
> 4. Major functionality bugs in newly-introduced features.
> 5. Regressions from older versions of Django.
> 
> The rule of thumb is that fixes will be backported to the last feature release for bugs that would have prevented a release in the first place (release blockers).

**Comment by timgraham** [comment]
Those backslashes have been in Django since 1.2 (https://github.com/django/django/commit/471596fc1afcb9c6258d317c619eaf5fd394e797#diff-62a68852d55ab3e8b0d79d74440bca1dR83) so unless something else changed, this doesn't look like a regression.

**Comment by maxnordlund** [comment]
I did, but I fail to see of this isn't considered eligible for inclusion. It fixes a bug, and the change in master is a breaking change while this is not (but probably better overall). So this is for the last Django still on 2.7 and 3.4 which will still be relevant for a fair amount of time to come, yes?

Yes, it true that it has been around for a long time, but I suspect that hasn't been uncovered before because it's one of two alternative paths, `<dot-atom>|<quoted-string>`.

Perhaps I'm being home blind here, but if some code is broken, why shouldn't it be fixed?

**Comment by timgraham** [comment]
The policy priorities stability to avoid the risk of regressions in the stable versions. If you need this fix, you can certainly monkeypatch your own projects.

**Comment by maxnordlund** [comment]
I see, well that makes sense. I'm pretty sure this doesn't introduce a regression, since all the tests are passing, both old and new. It _does_ accept a couple of more emails that are valid according to  RFC 5322 which weren't accepted previously.

I guess the only way to be more confident is to run this on a larger sample of real world email addresses.

**Comment by maxnordlund** [comment]
Whatever, at least I tried.

**Comment by maxnordlund** [comment]
After thinking about this a bit more I wanted to explain what I experienced trying to fix a bug. It's my personal opinion, and I'm not trying to blame anyone, just inform what it's like to be a complete newcomer.

**TL;DR** It doesn't feel like you want new contributors.

It's not easy to contribute to this project as it has a non trivial amount of red tape to wade through. Two bug trackers, a whole section of contributing guidelines, policy documents, CLA, … All very reasonable things, but it's a bit much, especially in the beginning.

When someone new takes the time to try to fix a bug, including tests, it would be nice if you get an decent explanation. The Django project has a _lot_ of documentation, and it's easy to miss things like the policy document linked above.

> @maxnordlund please take the time to read the docs Tim linked to.

@charettes I did read the policy, but it's not very clear what bug fixes should be included or not. Assuming that I didn't read what was just linked is quite patronising. I know you ment well, but writing always comes out harsher then what one might think, and I just wanted to let you know how it was received.

> The policy priorities stability to avoid the risk of regressions in the stable versions. If you need this fix, you can certainly monkeypatch your own projects.

@timgraham That sentence explains the policy much better then the current page, IMO. As for monkeypatching, that seems like an excellent way of shooting yourself in the foot, yes? From my limited understanding, isn't that frowned upon by the python community? (Coming from JS/Ruby where that's the case.)

In the end I gave up, and I wonder how many new contributors does the same, or who doesn't even try.

**Comment by charettes** [comment]
Thanks for your honest feedback @maxnordlund and sorry for the patronising comment. I can assure you I was under the assumption you missed the link @timgraham pointed to given the questions you were asking in your reply.

That might have been caused by my familairity with the backporting guidelines and the large number of requests we get to _fix_ how we validate emails.

**Comment by timgraham** [comment]
If you would like to propose some clarification to the supported versions policy, we can take a look. It's not clear to me what your confusion is. In past experience, providing that link is usually a sufficient explanation for contributors. The point is that this fix doesn't fall into any of the listed categories, therefore it doesn't qualify for a backport:
- Security issues.
- Data loss bugs.
- Crashing bugs.
- Major functionality bugs in newly-introduced features.
- Regressions from older versions of Django.

Sure, monkey patching is not a great solution but I was trying to be helpful since this issue seems really important to you.

Sorry if my reply seemed curt -- it's only due to busyness, not malice.
